BSP/Simulator/Src/BSP_CAN.c

```diff
@@ -131,24 +135,18 @@ uint8_t BSP_CAN_Read(CAN_t bus, uint32_t* id, uint8_t* data) {
     // Clear entries from file
     freopen(FILE_NAME, "w", fp);
 
-    // Re-write entry that wasn't read
-    // Leave entry that was read as a blank line
-    for (uint8_t i = 0; i < NUM_CAN; i++) {
-            if (bus == i) {
-                fprintf(fp, "\n");
-            } else {
-                // As long as it wasn't empty or a blank line,
-                // Print back data, otherwise, print newline
-                if((strcmp(currentCAN[i], "/n")!=0) && (strcmp(currentCAN[i], "")!=0)){
-                fprintf(fp, "%s", currentCAN[i]);
-                }
-                else{
-                    fprintf(fp, "\n");
-                }
-            }
-        }
+    // Rewrite unread entries
+    // Write newline for read entry and any entry that should be there but isn't
+    uint8_t index = 0;
+    for (; index < actualNumCan; index++) {
+        fprintf(fp, "%s", bus == index ? "\n" : currentCAN[index]);
+    }
+    for (; index < NUM_CAN; index++) {
+        fprintf(fp, "\n");
+    }
```

Comment:
Did you test that this works properly?

Reply:
Yes, I ran it with an empty file, a file with just one CAN line having data (in either position), and both CAN lines having data. Seemed to work for all of them on my end